### ISSUE TITLE
Username colors changed out of nowhere

### ISSUE THREAD

**Issue by WolfspiritM** [vector-im/element-web#26170]
### Steps to reproduce

Compare old and new username colors for example for userid "@sometest:matrix.org".

![image](https://github.com/vector-im/element-web/assets/5904171/a40389b2-3918-4b83-9e11-aadea817c0a0)

#### What did you expect?
Username colors staying the same

#### What happened instead?
Username colors changed drastically

As mention in another issue:
My color is completely different now and in conversations I always think I've a new message.
Especially confusing is that a friend has now a similar color I had before and I have now a color he had before.
Blue just became purple and purple became blue for example...

I always think I got a new message cause I got used to my color...

The reason is that the hashcode way that was implemented before worked differently.
(I still hope that there will be a way to set a custom color someday... (for everyone to see))

I think the old way was better cause it prevented someone with "misspelled" username to impersonate someone else. The new way produces the same color no matter in what order the letters are:

![image](https://github.com/vector-im/element-web/assets/5904171/95d08e37-b942-45db-b853-191aa9fd62a7)
![image](https://github.com/vector-im/element-web/assets/5904171/38e595df-a99a-4448-a3b2-5eb59524fb01)

What makes me believe that this is a bug is that the name of the function doesn't do what it suggests:

https://github.com/vector-im/compound-web/blob/1b0bf41cf7782db098eb67b9e4de683a2fb32eac/src/components/Avatar/useIdColorHash.ts#L23C20-L23C20

It's not using a "hash" anymore. The function that got removed instead used a hash to do that:
https://github.com/matrix-org/matrix-react-sdk/commit/e3df6624d68f736300777c3e6f7c7cd7a2d60e27#diff-1f693496f32ed3d0423cbc77412a1febdd04e3b0983a98ef06f8cd81262c8f9dL83

I think this is related to
https://github.com/vector-im/element-web/issues/26042

### Operating system

Windows

### Application version

started with v1.11.41

### Will you send logs?

No

**Comment by t3chguy** [comment]
> It's not using a "hash" anymore.

Yes it is

A hash is:

> assign a [numeric](https://www.google.com/search?sca_esv=565398918&q=numeric&si=ALGXSlYwkgxr-HbbJwcOTTqB6ethrjhYqmm5HmIJbPl0CzGDwmBI82fCWX2mSeZfkwUzzKFfA138zmi4DAA-qYwlwuj0krLNNA%3D%3D&expnd=1) or [alphanumeric](https://www.google.com/search?sca_esv=565398918&q=alphanumeric&si=ALGXSlb91IXEiYApD91csfAularihmrbnJWssCKEIxdXa2M9ja4fKm3gJpAbp8N0cnR90N7zn-UGl5gcrpHmafUJ6aQXZqvLaZoUOu4EeP_2Dbc56LbSCFc%3D&expnd=1) string to (a piece of data) by applying a function whose output values are all the same number of bits in length.

Given how Javascript represents numbers, this matches. It is a one-way function which maps a string to a number.

> The new way produces the same color no matter in what order the letters are:

Fair, but this isn't a requirement for a hash function to be a valid function.

I suggest an issue on compound if you have an issue with its specific implementation. 

https://github.com/vector-im/element-meta/issues/2046 is the issue about the new colours

**Comment by WolfspiritM** [comment]
> > The new way produces the same color no matter in what order the letters are:
> 
> Fair, but this isn't a requirement for a hash function to be a valid function.

I didn't count that as a requirement for a hash function and I agree that summing up the char codes is some kind of "hash" aswell (even a very poor man one). It just felt like it was meant to be implemented the same way but somehow the part about the "ORing" which made the result less predictable (and yes, I know that is not a requirement for a hash aswell) got removed.

I added my comment to the other issue.
